flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/MetadataOutputStreamWrapper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.filesystem;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.fs.FSDataOutputStream;
+
+import java.io.IOException;
+
+/** The wrapper manages metadata output stream close and commit. */
+@Internal
+public abstract class MetadataOutputStreamWrapper {
+    private volatile boolean closed = false;
+
+    /** Return {@link FSDataOutputStream} to write and other operations. */
+    abstract FSDataOutputStream getOutput();
+
+    /**
+     * The abstract function of once closing output stream and committing operation. It will throw
+     * {@link IOException} when failed and should be invoked by {@code closeForCommit()} indirectly
+     * instead of this function.
+     */
+    protected abstract void closeForCommitAction() throws IOException;
```

Comment:
Remove `protected` for these two methods to keep consistency?

Reply:
Make sense.